programs/monaco_protocol/src/instructions/math.rs

```diff
@@ -25,10 +27,13 @@ pub fn calculate_for_payout(stake: u64, price: f64) -> u64 {
     Decimal::from(stake).mul(price_decimal).to_u64().unwrap()
 }
 
-pub fn stake_precision_is_within_range(stake: u64, decimal_limit: u8) -> bool {
-    Decimal::new(stake as i64, decimal_limit as u32)
-        .fract()
-        .is_zero()
+pub fn stake_precision_is_within_range(stake: u64, decimal_limit: u8) -> Result<bool> {
+    let mut stake_decimal = Decimal::from_u64(stake).unwrap();
+    require!(
+        stake_decimal.set_scale(decimal_limit as u32).is_ok(),
+        CoreError::ArithmeticError
+    );
+    Ok(stake_decimal.fract().is_zero())
```

Comment:
how about this cheeky suggestion 😜 
```suggestion
pub fn stake_precision_is_within_range(stake: u64, decimal_limit: u8) -> bool {
    Decimal::from_i128_with_scale(stake as i128, decimal_limit as u32)
        .fract()
        .is_zero()
```